crates/protocol/src/block_info.rs

```diff
@@ -500,6 +516,38 @@ impl L1BlockInfoBedrock {
             l1_fee_scalar,
         })
     }
+
+    /// Calculate the data gas for posting the transaction on L1. Calldata costs 16 gas per byte
+    /// after compression.
+    ///
+    /// Prior to fjord, calldata costs 16 gas per non-zero byte and 4 gas per zero byte.
+    ///
+    /// Prior to regolith, an extra 68 non-zero bytes were included in the rollup data costs to
+    /// account for the empty signature.
+    pub fn data_gas(&self, input: &[u8]) -> U256 {
+        let mut rollup_data_gas_cost = U256::from(input.iter().fold(0, |acc, byte| {
+            acc + if *byte == 0x00 { ZERO_BYTE_COST } else { NON_ZERO_BYTE_COST }
+        }));
+
+        // Prior to regolith, an extra 68 non zero bytes were included in the rollup data costs.
+        rollup_data_gas_cost += U256::from(NON_ZERO_BYTE_COST).mul(U256::from(68));
+
+        rollup_data_gas_cost
+    }
```

Comment:
This is incorrect - the `data_gas` function in `L1BlockInfoBedrock` needs a separate variant for pre-Regolith to add the signature cost overhead (`NON_ZERO_BYTE_COST * 68`). After regolith, this value was removed from the cost.

Reply:
Is line 533 not doing this? Or maybe I'm not understanding

Reply:
It is performing the pre-regolith behavior, but there's nowhere to enable the regolith fork's behavior, which uses this variant of the L1 info transaction. The specs for the change are [here](https://specs.optimism.io/protocol/exec-engine.html#pre-ecotone), see "Before Regolith fork..." and "With Regolith fork..."